src/CouchDB.Driver/Translators/ConstantExpressionTranslator.cs

```diff
@@ -41,14 +41,16 @@ protected override Expression VisitConstant(ConstantExpression c)
                             VisitIEnumerable(c.Value as IList<int>);
                         else if (c.Value is IList<long>)
                             VisitIEnumerable(c.Value as IList<long>);
-                        else if(c.Value is IList<decimal>)
+                        else if (c.Value is IList<decimal>)
                             VisitIEnumerable(c.Value as IList<decimal>);
                         else if (c.Value is IList<float>)
                             VisitIEnumerable(c.Value as IList<float>);
                         else if (c.Value is IList<double>)
                             VisitIEnumerable(c.Value as IList<double>);
-                        else if(c.Value is IList<string>)
+                        else if (c.Value is IList<string>)
                             VisitIEnumerable(c.Value as IList<string>);
+                        else if (c.Value is Guid)
+                            _sb.Append(JsonConvert.SerializeObject(c.Value));
                         else
                             throw new NotSupportedException(string.Format("The constant for '{0}' is not supported", c.Value));
```

Comment:
@matteobortolazzo I was tempted to just trust Json.Net and `_sb.Append(JsonConvert.SerializeObject(c.Value));` for the unknown case instead of throwing. What are your thoughts on that?

Reply:
Guid is nice to have!
As for the unknown case we can try.
I'm going to merge it later.